lib/cocoapods-core/source/aggregate.rb

```diff
@@ -198,26 +187,34 @@ def update_search_index(search_data)
       # @param  [Set] set
       #         The set for which the information is needed.
       #
-      # @note   If the specification can't load an empty hash is returned and
+      # @note   If the specification can't load an empty array is returned and
       #         a warning is printed.
       #
       # @note   For compatibility with non Ruby clients a strings are used
       #         instead of symbols for the keys.
       #
-      # @return [Hash{String=>String}] A hash with the search information.
+      # @return [Array<String>] An array of words contained by the set's search related information.
       #
-      def search_data_from_set(set)
-        result = {}
+      def word_list_from_set(set)
         spec = set.specification
-        result['version'] = spec.version.to_s
-        result['summary'] = spec.summary
-        result['description'] = spec.description
-        result['authors'] = spec.authors.keys.sort * ', '
-        result
+        string = set.name.dup
+        if spec.summary
+          string << ' ' << spec.summary
```

Comment:
In Ruby, it's generally better to avoid `string << ' ' << spec.summary` because with each `<<`, a String is created. Use arrays to `<<` the strings, then `join` once at the end.

Or better yet, in this case, only split where necessary (e.g. in summary) and append the result to the array using `+=`, then `uniq` at the end.